core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.cassandra.query
+
+import java.lang.{ Long => JLong }
+import java.nio.ByteBuffer
+import java.util.concurrent.ThreadLocalRandom
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.persistence.PersistentRepr
+import akka.persistence.cassandra.ListenableFutureConverter
+import akka.persistence.cassandra.journal.CassandraJournal
+import akka.serialization.SerializationExtension
+import akka.stream.ActorMaterializer
+import akka.stream.Attributes
+import akka.stream.Outlet
+import akka.stream.SourceShape
+import akka.stream.stage.GraphStageLogic
+import akka.stream.stage.GraphStageWithMaterializedValue
+import akka.stream.stage.OutHandler
+import akka.stream.stage.StageLogging
+import akka.stream.stage.TimerGraphStageLogic
+import com.datastax.driver.core.ConsistencyLevel
+import com.datastax.driver.core.PreparedStatement
+import com.datastax.driver.core.ResultSet
+import com.datastax.driver.core.Row
+import com.datastax.driver.core.Session
+import com.datastax.driver.core.Statement
+import com.datastax.driver.core.policies.RetryPolicy
+import com.datastax.driver.core.utils.Bytes
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object EventsByPersistenceIdStage {
+
+  // materialized value
+  trait Control {
+    /**
+     * Trigger a request to fetch more events.
+     */
+    def poll(knownSeqNr: Long): Unit
```

Comment:
I guess `knownSeqNr` means in that case, that the caller of this method already learned about new sequence numbers being available through some other channel.

Reply:
yes, we are currently only using that to check if the poll is still relevant with `knownSeqNr >= seqNr`, but one could imagine more advanced things in the future such as requesting db more frequently until the `knownSeqNr` arrives